DNN Platform/Modules/HTML/Components/WorkflowStateController.cs

```diff
@@ -13,16 +13,31 @@ namespace DotNetNuke.Modules.Html
     public class WorkflowStateController
     {
         private const string WORKFLOWCACHEKEY = "Workflow{0}";
+        private const string WORKFLOWSCACHEKEY = "Workflows{0}";
         private const int WORKFLOWCACHETIMEOUT = 20;
 
         private const CacheItemPriority WORKFLOWCACHEPRIORITY = CacheItemPriority.Normal;
 
-        /// <summary>GetWorkFlows retrieves a collection of workflows for the portal.</summary>
+        /// <summary>GetWorkFlows retrieves a collection of workflows for the portal from the cache.</summary>
         /// <param name="portalID">The ID of the Portal.</param>
         /// <returns>An <see cref="ArrayList"/> of <see cref="WorkflowStateInfo"/> instances.</returns>
         public ArrayList GetWorkflows(int portalID)
         {
-            return CBO.FillCollection(DataProvider.Instance().GetWorkflows(portalID), typeof(WorkflowStateInfo));
+            var cacheKey = string.Format(WORKFLOWSCACHEKEY, portalID);

```

Comment:
Do we need to also clear this cache if a workflow gets updated/added/removed?

Reply:
Probably, but I haven't found any code that does such a thing...

Reply:
@bdukes  Would it make sense to add an optional clearCache parameter to the method so extensions can control the cache themselves? Or a new method "ClearWorkflowsCache" or something?

Reply:
You're right, there isn't any code to modify these HTML workflows (in contrast to the content workflows used by the platform). Clearing the cache would be the responsibility of whoever is updating that table. I'm okay with this as-is.